### ISSUE TITLE
Ability to add fonts manually

### ISSUE THREAD

**Issue by corecode1** [felixmariotto/three-mesh-ui#42]
It would be nice to be able to add fonts manually instead of loading when requested. Some apps prefer loading all assets prior to starting expirience (this is my case), and yet this is not possible with three-mesh-ui.

It seems like easy-to implement feature to me, so I could make a pull request with it, but i'm new to this library, and not sure which api would be the best choice. My current suggestion is to add some method like `addFont(json, texture)` to FontLibrary and expose FontLibrary to ThreeMeshUI object.

**Comment by felixmariotto** [comment]
Hi @corecode1, yes I've not thought about this workflow but I would gladly accept a PR adding support for it.

About the API :  

The Text component architecture is built to make it easy to add a custom Text component, because this library made the choice of using [MSDF](https://github.com/Chlumsky/msdfgen)-based text, but there had been discussions about other types of text, which could prove more suited to some cases. Therefore the `addFont` method should be Text-type agnostic and take a texture _optionally_.

Also [FontLibrary](https://github.com/felixmariotto/three-mesh-ui/blob/master/src/components/core/FontLibrary.js) stores the fonts in `requiredFontTextures` and `requiredFontFamilies `with the url of the font file as key. If the files are pre-loaded, a name should be provided by the user, so the objects can be accessed with the name of the font as key : 
```javascript
const block = new ThreeMeshUI.Block({
	fontFamily: 'name-of-my-font',
	fontTexture: 'name-of-my-font'
});
```

So the API should be : `addFont( name, json, texture )` with the texture argument optional.

Thanks a lot for your interest in this library 😊

**Comment by corecode1** [comment]
Hi @felixmariotto , thanks for the responce! I've just created a pull reqeuest with the feature implemented. API looks just  the way you described. Implementation has straight-forward approach, `addFont` method assumes no fonts with such name have been requested or created yet, and it doesen't update components. 

I also made new example that shows how to use new feature. It is just a copy of basic setup example with the only difference that font is loaded beforehead. However, only texture is preloaded in the example, and font is embedded in the script. Currently webpack seems to be configured not to serve jsons from `examples/assets/.` There were some errors when i tried to serve jsons as well, so for now i just left it as is.
